binarytree/__init__.py

```diff
@@ -136,24 +136,22 @@ def _build_tree(values):
     root = _new_node(values[0])
     nodes[0] = root
 
-    index = 1
-    while index < len(values):
-        value = values[index]
-        if value != _null:
-            parent_index = int((index + 1) / 2) - 1
-            parent_node = nodes[parent_index]
-            if parent_node == _null:
-                raise ValueError(
-                    'Node missing at index {}'
-                    .format(parent_index)
-                )
-            child_node = _new_node(value)
-            if index % 2:  # is odd
-                _add_left(parent_node, child_node)
-            else:
-                _add_right(parent_node, child_node)
-            nodes[index] = child_node
-        index += 1
+    for index, value in enumerate(values):
+        if value == _null or index == 0:
```

Comment:
Hmm this doesn't seem very elegant as we are adding an unnecessary check every iteration. I don't see a compelling reason to use `enumerate` here. We can keep the continue as I've said before to reduce indentation!